File: backend-src/UZonMailService/Models/SqlLite/Settings/UserSetting.cs
using UZonMailService.Models.SqlLite.Base;

namespace UZonMailService.Models.SqlLite.Settings
{
    /// <summary>
    /// 用户设置
    /// </summary>
    public class UserSetting : SqlId
    {
        /// <summary>
        /// 用户 id
        /// </summary>
        public int UserId { get; set; }

        /// <summary>
        /// 每日每个发件箱最大发送次数
        /// 为 0 时表示不限制
        /// </summary>
        public int MaxSendCountPerEmailDay { get; set; } = 0;

        /// <summary>
        /// 最小发件箱冷却时间
        /// </summary>
        public int MinOutboxCooldownSecond { get; set; } = 5;

        /// <summary>
        /// 最大发件箱冷却时间
        /// </summary>
        public int MaxOutboxCooldownSecond { get; set; } = 10;

        /// <summary>
        /// 最大批量发件数
        /// </summary>
        public int MaxSendingBatchSize { get; set; } = 20;

        /// <summary>
        /// 获取冷却时间
        /// 随机
        /// </summary>
        /// <returns></returns>
        public int GetCooldownMilliseconds()
        {
            if(MinOutboxCooldownSecond == MaxOutboxCooldownSecond)
            {
                return MinOutboxCooldownSecond * 1000;
            }

            // 随机从 min 到 max 取值
            return new Random().Next(MinOutboxCooldownSecond, MaxOutboxCooldownSecond) * 1000;
        }
    }
}

File: ui-src/src/api/userSetting.ts
/* eslint-disable @typescript-eslint/no-explicit-any */
import { httpClient } from 'src/api//base/httpClient'

export interface IUserSetting {
  userId: string,
  maxSendCountPerEmailDay: number,
  minOutboxCooldownSecond: number,
  maxOutboxCooldownSecond: number,
  maxSendingBatchSize: number,
  minInboxCooldownHours: number,
  replyToEmails?: string
}

/**
 * 获取用户设置
 * @returns
 */
export function getCurrentUserSetting () {
  return httpClient.get<IUserSetting>('/user-setting')
}

/**
 * 更新用户设置
 * @param userSetting
 * @returns
 */
export function updateUserSetting (userSetting: IUserSetting) {
  return httpClient.put<Record<string, any>>('/user-setting', {
    data: userSetting
  })
}
